### PR TITLE
3.0.0 RC.11

### PR DESCRIPTION
**Pull request by robonaut**

### CODE REVIEW

**Comment by ralf601** [comment]
@robonaut have you compiled this?  it's not compiling due to a merge from develop.

File: android/build.gradle [hunk 1]
```diff
@@ -45,7 +45,7 @@ repositories {
 
 dependencies {
     implementation 'com.facebook.react:react-native:+'
-    worldApi('com.sentiance:sdk:4.10.0-R@aar') {transitive = true}
+    implementation ('com.sentiance:sdk:4.10.0@aar') { transitive = true }
```
Comment (ralf601) [inline android/build.gradle]: any reason for changing worldApi -> implementation? for me its not picking the correct variant if not specified in gradle file.

File: ios/RNSentiance.m [hunk 2]
```diff
@@ -131,18 +197,6 @@ - (void)initializeSDK:(NSString *)appId secret:(NSString *)secret baseURL:(NSStr
     }
 }
 
-
-//Deprecated method use -(SENTSDKInitState)getInitState;
-RCT_EXPORT_METHOD(isInitialized:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
```
Comment (ralf601) [inline ios/RNSentiance.m]: i think its better to keep this method and annotate with depreciation, if someone is already using it code wont compile.

Comment (robonaut) [reply ios/RNSentiance.m]: The reason we do a major semver release is because we have breaking changes.

Comment (robonaut) [reply ios/RNSentiance.m]: Added [CHANGELOG](https://github.com/sentiance/react-native-sentiance/pull/83/commits/3378481534f6e4572529dce8709992020ad7d792) to start documenting releases

Comment (ralf601) [reply ios/RNSentiance.m]: remove this from android also to make it consistent.

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import { EventSubscriptionVendor } from "react-native";
+
+declare module "react-native-sentiance" {
+  interface RNSentianceConstructor extends EventSubscriptionVendor {
+    init(appId: string, secret: string, baseURL: string): Promise<any>;
```
Comment (jiricrispeyn) [inline index.d.ts]: Not blocking but can we try to replace the `any` with a more detailed return value type?

File: android/src/main/java/com/sentiance/react/bridge/RNSentianceHelper.java [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+    });
+  }
+  @SuppressWarnings({"unused","WeakerAccess"})
+  public void startSentianceSDK(final long stopDateEpoch, @Nullable final OnStartFinishedHandler callback) {
+    Context context = weakContext.get();
+    if(context==null) return;
+    Sentiance.getInstance(context).start(new Date(stopDateEpoch),new OnStartFinishedHandler() {
+      @Override
+      public void onStartFinished(SdkStatus sdkStatus) {
+        if(callback !=null)
```
Comment (jiricrispeyn) [inline android/src/main/java/com/sentiance/react/bridge/RNSentianceHelper.java]: Can we make sure code formatting is consistent throughout the source code?

File: android/build.gradle [hunk 5]
```diff
@@ -45,7 +45,7 @@ repositories {
 
 dependencies {
     implementation 'com.facebook.react:react-native:+'
-    worldApi('com.sentiance:sdk:4.10.0-R@aar') {transitive = true}
+    worldApi ('com.sentiance:sdk:4.10.0@aar') { transitive = true }
     chinaApi('com.sentiance:sdk-cn:4.5.0-CN@aar') {
```
Comment (jiricrispeyn) [inline android/build.gradle]: With these updates and the CN version still being on `4.5.0-CN`, it's not going to compile on this flavour.
Updating the SDK is something that won't be happening any time soon I guess, so the options we currently have are:
- Remove the support (will break some apps that have integrated the CN SDK with the RN bridge)
- Add info about CN support to `README.md`
- Figure out a way to support both flavours through our bridge

File: android/build.gradle [hunk 6]
```diff
@@ -7,19 +7,19 @@ buildscript {
...
-    buildToolsVersion "27.0.3"
+    buildToolsVersion "28.0.3"
     defaultPublishConfig "worldRelease"
     defaultConfig {
         minSdkVersion 16
         targetSdkVersion 26
-        versionCode 13
-        versionName "2.7"
+        versionCode 18
+        versionName "4.0"
```
Comment (jiricrispeyn) [inline android/build.gradle]: Any reason why we skip `3.X`?

File: CHANGELOG.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
+# Changelog
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
+
+## [3.0.0] - 2019-08-01
+### Added
```
Comment (jiricrispeyn) [inline CHANGELOG.md]: Maybe we need to add some info about adding support for user activity over here?

File: android/build.gradle (hunk 5 above)
Comment (ralf601) [reply android/build.gradle]: we can use **4.7.0-CN** which is equivalent to **4.10.0** global then code will compile.

**Pull request merged**